tests/unit/core/metrics/test_confusion_matrix.py

```diff
@@ -10,6 +12,36 @@
 from whylogs.proto import ScoreMatrixMessage
 
 
+def _generateTestTargetsAndPredictions(label_size: int, output_length, rand_ratio: float):
+    labels = [i for i in range(label_size)]
+    targets = []
+    predictions = []
+
+    for _ in range(output_length):
+        label = random.choice(labels)
+        targets.append(label)
+        if random.random() < rand_ratio:
+            predictions.append(random.choice(labels))
+        else:
+            predictions.append(label)
+    return targets, predictions
+
+
+def _generateTestTargetsAndPredictionsForEachLabel(label_size: int, output_length, rand_ratio: float):
+    labels = [i for i in range(label_size)]
+    targets = []
+    predictions = []
+
+    for i in range(output_length):
+        label = labels[i % label_size]
```

Comment:
Pass label selector in as a callable so these two can be unified?

Reply:
I removed one of the methods for now, but good suggestion.